docs/DeployOnAzure.md

```diff
@@ -0,0 +1,295 @@
+# Eclipse Hono deployment on Microsoft Azure
```

Comment:
we maintain the deployment guides in the `site/content/deployment/` folder. Can you please move this page there? I will then add a corresponding header.

Reply:
Hi @sophokles73 , Thank you for your comment. Added commit with those changes: https://github.com/eclipse/hono/pull/1117/commits/f80793136cf65c1f286a54d53a7163f9d66c5dea